### PR TITLE
Fixing support for mingw toolchains that target the newer ucrt

### PR DESCRIPTION
**Pull request by Honeybunch**
Attempting to fix: https://github.com/KhronosGroup/KTX-Software/issues/641

### CODE REVIEW

File: lib/filestream.c [hunk 1]
```diff
@@ -307,13 +307,13 @@ KTX_error_code ktxFileStream_getsize(ktxStream* str, ktx_size_t* size)
...
-    // Can ignore return value. The only error that can happen is to tell you
-    // it was a NOP because the file is read only.
-#if (defined(_MSC_VER) && _MSC_VER < 1900) || defined(__MINGW32__)
-    // Bug in VS2013 msvcrt. fflush on FILE open for READ changes file offset
-    // to 4096.
-    if (str->data.file->_flag & _IOWRT)
+  // Need to flush so that fstat will return the current size.
+  // Can ignore return value. The only error that can happen is to tell you
+  // it was a NOP because the file is read only.
+#if (defined(_MSC_VER) && _MSC_VER < 1900) || defined(__MINGW32__) && !defined(_UCRT)
```
Comment (MarkCallow) [inline lib/filestream.c:313]: Should we change `__MINGW32__` to `__MINGW64__`?

Comment (Honeybunch) [reply lib/filestream.c:313]: Oh yeah I forgot to reply to that part of the issue. I don't think we really need to. It'll work the same either way but I can change it if you think it's a good idea.

Comment (MarkCallow) [reply lib/filestream.c:313]: Please change it to `__MINGW64__`. Since KTX-Software only supports 64-bit builds, it will be less confusing.

Comment (MarkCallow) [reply lib/filestream.c]: That was quick! Thanks.

**Pull request merged**